codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/JsonSerdeDescriptorGenerator.kt

```diff
@@ -24,6 +25,73 @@ open class JsonSerdeDescriptorGenerator(
     private val supportsJsonNameTrait: Boolean = true,
 ) : AbstractSerdeDescriptorGenerator(ctx, memberShapes) {
 
+    override fun render() {
```

Comment:
**fix**: This logic is specific to AWS protocols and doesn't belong in `smithy-kotlin`